commands/command_integrate.go

```diff
@@ -0,0 +1,206 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/pflag"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/spf13/cobra"
+)
+
+var (
+	fileTemplate1 = `
+
+_git-lfs() {
+  local -a commands help
+  commands=(
+`
+	fileTemplate2 = `)
+
+  help=(
+  'help:Command help'
+  )
+
+  _arguments \
+    "1: :{_describe 'command' commands -- help}" \
+    "*:: :->args"
+
+  case $state in
+    args)
+      case $words[1] in
+`
+	fileTemplate3 = `
+        help)
+          _arguments \
+            "1: :{_describe 'command' commands}"
+          ;;
+      esac
+      ;;
+  esac
+}
+
+compdef _git-lfs git-lfs
+`
+	integrateCmd = &cobra.Command{
+		Use:   "integrate",
+		Short: "Integrate git-lfs with external tools",
+		Run:   integrateCommand,
+	}
+
+	zshOutputFile  string
+	zshOhMyZshCore bool
+)
+
+func getHomeDir() string {
+	usr, err := user.Current()
+	if err != nil {
+		Exit("Unable to locate home directory: %v", err)
+	}
+	return usr.HomeDir
+}
+
+func getOhMyZshRootDir() string {
+	// Can't rely on ZSH_CUSTOM etc because it's only defined inside zsh shell, it's
+	// not actually a proper environment var visible in child processes
+	zsh := os.Getenv("ZSH")
+	if len(zsh) > 0 {
+		// Check that it is ohmyzsh
+		testfile := filepath.Join(zsh, "oh-my-zsh.sh")
+		if _, err := os.Stat(testfile); err == nil {
+			return zsh
+		}
+	}
+
+	return ""
+}
+
+func integrateCommand(cmd *cobra.Command, args []string) {
+
+	if len(args) == 0 {
+		Exit("Required: tool argument")
+	}
+
+	switch args[0] {
+	case "zsh":
+		zshCommand(args[1:])
+	}
+}
+
+func zshCommand(args []string) {
+	var outputFile string
+
+	if len(zshOutputFile) > 0 {
+		outputFile = zshOutputFile
+	} else if zshOhMyZshCore {
+		// Write to core oh-my-zsh for submission upstream
+		ohmyzsh := getOhMyZshRootDir()
+		if len(ohmyzsh) == 0 {
+			Exit("Could not locate oh-my-zsh root dir via $ZSH")
+		}
+		outputFile = filepath.Join(ohmyzsh, "plugins", "git-lfs", "git-lfs.plugin.zsh")
+	} else if ohmyzsh := getOhMyZshRootDir(); len(ohmyzsh) > 0 {
+		// default user-specific oh-my-zsh
+		outputFile = filepath.Join(getOhMyZshRootDir(), "custom", "plugins", "git-lfs", "git-lfs.plugin.zsh")
+	} else {
+		outputFile = filepath.Join(getHomeDir(), "_git-lfs.zsh")
```

Comment:
It doesn't have to be, you can just `source` it in your `~/.zshrc. It can auto-load if you put it in `$fpath` and it's called `_git-lfs` but I have no way to tell which folder in `$fpath` to put it in; I'd have to put it somewhere else and maybe add it to `$fpath` which I didn't really like either. Thought the best option was to let the user decide & just tell them where it is.